aux_commands/assign_group.py

```diff
@@ -0,0 +1,122 @@
+from typing import List, Dict
+
+import random
+import discord
+
+from aux_commands.create_delete_group import aux_create_group
+from aux_commands.join_leave_group import aux_join_group
+from aux_commands.open_close_groups import is_open_group
+from utils import bot_messages as btm, helper_functions as hpf
+from utils.guild_config import GUILD_CONFIG
+
+"""
+####################################################################
+#################### ASSIGN GROUP FUNCTIONS ########################
+####################################################################
+"""
+
+async def aux_assign_all(ctx):
+    print('Assigning students to open groups automatically')
+    # Get number of members per group
+    size_to_group = await map_size_of_members_to_group(ctx.guild)
+    largest_size = max(size_to_group)
+    max_group_size = GUILD_CONFIG.max_students_per_group(ctx.guild)
+
+    no_group_students = hpf.all_students_with_no_group(ctx.guild)
+    print(str(len(no_group_students))+' students are without a group')
+    online_no_group_students = hpf.select_online_members(ctx.guild, no_group_students)
+    # online_no_group_students = no_group_students   # ... for debugging late at night
+    print(str(len(online_no_group_students))+' students are online without a group')
+
+    online_no_group_students_with_nick = []
```

Comment:
why is it necessary to give students with nick a different treatment? IIRC I had a way to handle both cases (with nick and without it)